packages/1155-contracts/package/premint-api.ts

```diff
@@ -0,0 +1,330 @@
+import { createPublicClient, http } from "viem";
+import type {
+  Account,
+  Address,
+  Chain,
+  Hex,
+  PublicClient,
+  WalletClient,
+} from "viem";
+import {
+  zoraCreator1155PremintExecutorImplABI,
+  zoraCreator1155PremintExecutorImplAddress,
+  zoraCreatorFixedPriceSaleStrategyAddress,
+} from "./wagmiGenerated";
+import { foundry, zora, zoraTestnet } from "viem/chains";
+import { PremintConfig, preminterTypedDataDefinition } from "./preminter";
+
+type NetworkConfig = {
+  chainId: number;
+  zoraPathChainName: string;
+  zoraBackendChainName: string;
+  isTestnet: boolean;
+};
+
+enum BackendChainNames {
+  ZORA_MAINNET = "ZORA-MAINNET",
+  ZORA_TESTNET = "ZORA-TESTNET",
+}
+
+const ZORA_API_BASE = "https://api.zora.co/premint/";
+
+const networkConfigByChain: Record<number, NetworkConfig> = {
+  [zora.id]: {
+    chainId: zora.id,
+    isTestnet: false,
+    zoraPathChainName: "zora",
+    zoraBackendChainName: BackendChainNames.ZORA_MAINNET,
+  },
+  [zoraTestnet.id]: {
+    chainId: zora.id,
+    isTestnet: true,
+    zoraPathChainName: "zora",
+    zoraBackendChainName: BackendChainNames.ZORA_TESTNET,
+  },
+  [foundry.id]: {
+    chainId: foundry.id,
+    isTestnet: true,
+    zoraPathChainName: "zora",
+    zoraBackendChainName: BackendChainNames.ZORA_TESTNET,
+  }
+};
+
+type MintArgumentsSettings = {
+  tokenURI: string;
+  maxSupply?: bigint;
+  maxTokensPerAddress?: bigint;
+  pricePerToken?: bigint;
+  mintStart?: bigint;
+  mintDuration?: bigint;
+  royaltyMintSchedule?: number;
+  royaltyBPS?: number;
+  royaltyRecipient?: Address;
+  fixedPriceMinter?: Address;
+};
+
+const OPEN_EDITION_MINT_SIZE = "18446744073709551615";
```

Comment:
how about making this a bigint, and exporting it?
`export const OPEN_EDITION_MINT_SIZE = 18446744073709551615n;`